File: database/src/main/java/bao/webapp/service/OrderServiceImpl.java
package bao.webapp.service;

import java.util.ArrayList;
import java.util.List;

import javax.transaction.Transactional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import bao.webapp.model.Order;
import bao.webapp.model.User;
import bao.webapp.repository.OrderRepository;

@Service
@Transactional
public class OrderServiceImpl implements OrderService {

	@Autowired
	private OrderRepository orderRepository;

	@Override
	public void create(Order order) {
		this.orderRepository.save(order);
	}

	@Override
	public void delete(Order order) {
		this.orderRepository.delete(order);
	}

	@Override
	public Order findById(long id) {
		return this.orderRepository.findById(id).get();
	}

	@Override
	public List<Order> findByUserID(long userID) {
		List<Order> all = new ArrayList<>();

		for (Order order : this.orderRepository.findByUserID(userID))
			all.add(order);

		return all;
	}

	@Override
	public Iterable<Order> all() {
		return this.orderRepository.findAll();
	}

}

File: database/src/main/kotlin/com/ossp/database/service/orders/OrderServiceImpl.kt
package com.ossp.database.service.orders

import com.ossp.database.error.ObjectNotFound
import com.ossp.database.model.Order
import com.ossp.database.repository.OrderRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import javax.transaction.Transactional

@Service
@Transactional
class OrderServiceImpl : OrderService {

    @Autowired
    private val orderRepository: OrderRepository? = null

    override fun create(order: Order): Order? {
        return this.orderRepository?.save(order)
    }

    override fun delete(order: Order?) {
        order?.let {
            this.orderRepository?.delete(it)
        }
    }

    override fun findById(id: Long?): Order? {
        id?.let {
            return this.orderRepository?.findById(it)?.get()
        } ?: run {
            throw ObjectNotFound(message = "Order Not Found")
        }
    }

    override fun findByUserID(userID: Long?): List<Order> {
        val all = ArrayList<Order>()

        orderRepository?.findByUserID(userID)?.let { orders ->
            for (order in orders) {
                all.add(order)
            }
        }

        return all
    }

    override fun all(): Iterable<Order>? {
        return this.orderRepository?.findAll()
    }

}
